### ISSUE TITLE
Parser failure reported in position of parent scope

### ISSUE THREAD

**Issue by fpoli** [fpoli/viper-silver#258]
The position parser failures is reported in the wrong places. For example, with different contexts above a while loop in a method implementation we get different locations for a parse error in the loop body. See screenshots and gist below. 

![Screen Shot 2016-07-28 at 2.46.01 PM.png](https://bitbucket.org/repo/8zGbGd/images/3171539484-Screen%20Shot%202016-07-28%20at%202.46.01%20PM.png)

![Screen Shot 2016-07-28 at 2.45.43 PM.png](https://bitbucket.org/repo/8zGbGd/images/1735388946-Screen%20Shot%202016-07-28%20at%202.45.43%20PM.png)

![Screen Shot 2016-07-28 at 2.45.29 PM.png](https://bitbucket.org/repo/8zGbGd/images/4261399639-Screen%20Shot%202016-07-28%20at%202.45.29%20PM.png)

```
method foo(s: Int) returns (x: Int)
{

    //assert true
    //x := s

    //assert true

    while (true) 
    {
        asset bar
        while (false) {
            karamba
        }
    }

}
```

**Comment by fpoli** [comment]
> **@alexander_summers** commented on 2019-01-22 16:34

**Comment by fpoli** [comment]
> **@martin_clochard** commented on 2019-02-27 14:54

Fixed in commit 6d6e69e
